### PR TITLE
Whisper preliminary implementation

### PR DESCRIPTION
**Pull request by mydatascience**
Trying to port Whisper model to NNX 
Includes the example for running  the Whisper model for the G. Bush speech transcription.

### CODE REVIEW

File: bonsai/models/whisper/tests/run_model_hf.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import os
```
[the remark this answers is not in the document]
Comment (mydatascience) [reply bonsai/models/whisper/tests/run_model_hf.py]: removed

File: bonsai/models/whisper/tokenizer.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+import os
+import string
+from dataclasses import dataclass, field
+from functools import cached_property, lru_cache
+from typing import Dict, List, Optional, Tuple, Union
+
+# Use HuggingFace tokenizer instead of tiktoken
+from transformers import WhisperTokenizer
+
+LANGUAGES = {
```
Comment (jenriver) [inline bonsai/models/whisper/tokenizer.py:16]: Do we need this? We can just call whichever language directly via `WhisperTokenizer(language=...)`

File: bonsai/models/whisper/tests/run_model.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        
+        # Get audio features
+        print(f"\n🎵 Computing audio features...")
+        audio_features = jax_model.embed_audio(mel_30s)
+        print(f"✅ Audio features computed - shape: {audio_features.shape}")
+        
+        # Generate tokens
+        # Initialize HuggingFace tokenizer
+        print(f"\n🔤 Initializing HuggingFace tokenizer...")
+        tokenizer_instance = get_tokenizer(multilingual=True, language="en", task="transcribe")
```
Comment (jenriver) [inline bonsai/models/whisper/tests/run_model.py:85]: We can just directly use `WhisperTokenizer.from_pretrained("openai/whisper-tiny", language="english", task="transcribe")` here.

If we're only using certain tokenizer imports (ex: tokenizer.sot, tokenizer.to_language_token("en"), tokenizer.transcribe, tokenizer.no_timestamps) once, can we just hardcode the variables (ex: `<|startoftext|>`) in this file? We want to make these implementations as simple and hackable as possible with minimal dependencies that are not directly used.


https://huggingface.co/docs/transformers/model_doc/whisper#transformers.WhisperTokenizer

File: bonsai/models/whisper/tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
+"""
```
Comment (jenriver) [inline bonsai/models/whisper/tokenizer.py:1]: Is this file a modue or a test file? I don't think we should have the test logic and function class together in the same file.

Also, I don't think most of the functions here are used (ex: `decode_with_timestamps`, etc).
Could we remove this file and call `WhisperTokenizer` directly? We'd like to not add an unnecessary abstraction, if possible.

File: bonsai/models/whisper/tests/audio_samples/bush_speech.wav
[the remark this answers is not in the document]
Comment (jenriver) [reply bonsai/models/whisper/tests/audio_samples/bush_speech.wav:1]: This seems to be still here, could we remove this?

**Comment by jenriver** [comment]
Can you also add a quality logits checker (ex: [test_outputs.py](https://github.com/jax-ml/bonsai/blob/dea803496c9073e0067da9d8ee97cae6466f8a7c/bonsai/models/vgg19/tests/test_outputs.py)) to verify that the resulting logits are correct?


Also, it seems some comments are yet to be addressed (ex: unnecessary key mapping in tokenizer, need to remove data files and instead refer via wget) -- please let us know when it is ready to be reviewed so that we can get whisper checked in :)

File: bonsai/models/whisper/tests/run_model.py (hunk 3 above)
Comment (mydatascience) [reply bonsai/models/whisper/tests/run_model.py]: Yes called it directly
